### PR TITLE
Fix/restriction tests merge

### PR DESCRIPTION
**Pull request by ravitej-aluru**

### CODE REVIEW

File: integrationtests/src/test/resources/io/nem/symbol/account_restriction/prevent_receiving_transactions_from_undesired_addresses.feature [hunk 1]
```diff
@@ -19,52 +19,39 @@ Feature: Prevent receiving transactions from undesired addresses
     When Alex tries to send 1 asset "cat.currency" to Bobby
     Then Alex should receive the error "FAILURE_RESTRICTIONACCOUNT_ADDRESS_INTERACTION_PROHIBITED"
 
-  Scenario: An account only allows receiving transactions from a set of addresses
-    Given Bobby only allowed receiving transactions from:
-      | Alex   |
-    When Carol tries to send 1 asset "cat.currency" to Bobby
-    Then Carol should receive the error "FAILURE_RESTRICTIONACCOUNT_ADDRESS_INTERACTION_PROHIBITED"
-
   Scenario: An account only allows receiving transactions from a set of addresses
```
Comment (Wayonb) [inline integrationtests/src/test/resources/io/nem/symbol/account_restriction/prevent_receiving_transactions_from_undesired_addresses.feature:22]: As discussed, we should start breaking up these into multiple scenarios

File: integrationtests/src/test/resources/io/nem/symbol/account_restriction/prevent_receiving_transactions_from_undesired_addresses.feature [hunk 2]
```diff
@@ -19,52 +19,39 @@ Feature: Prevent receiving transactions from undesired addresses
...
+    And Carol should receive the error "FAILURE_RESTRICTIONACCOUNT_ADDRESS_INTERACTION_PROHIBITED"
 
   Scenario: An account unblocks an address
     Given Bobby blocked receiving transactions from:
       | Alex  |
       | Carol  |
-    And Bobby removes Alex from blocked addresses
-    When Alex sends 1 asset "cat.currency" to Bobby
-    Then Bobby should receive 1 of asset "cat.currency"
+    When Bobby removes Alex from blocked addresses
```
Comment (Wayonb) [inline integrationtests/src/test/resources/io/nem/symbol/account_restriction/prevent_receiving_transactions_from_undesired_addresses.feature:36]: same here.

File: integrationtests/src/test/resources/io/nem/symbol/account_restriction/prevent_sending_transactions_by_type.feature [hunk 3]
```diff
@@ -8,85 +8,56 @@ Feature: Prevent sending transactions by type
       | Alex  |
       | Bobby |
       | Carol |
-    And the following transaction types are available:
+    Given the following transaction types are available:
       | TRANSFER                       |
-      | NAMESPACE_REGISTRATION         |
+      | REGISTER_NAMESPACE             |
```
Comment (Wayonb) [inline integrationtests/src/test/resources/io/nem/symbol/account_restriction/prevent_sending_transactions_by_type.feature:13]: REGISTER_NAMESPACE got rename to NAMESPACE_REGISTRATION

File: integrationtests/src/test/resources/io/nem/symbol/account_restriction/prevent_sending_transactions_by_type.feature [hunk 4]
```diff
@@ -8,85 +8,56 @@ Feature: Prevent sending transactions by type
...
       | ACCOUNT_PROPERTIES_ENTITY_TYPE |
 
     # We are using three transaction types for better comprehension.
     # To get all the available transaction types, see the NEM Developer Center/Protocol/Transaction.
 
-  Scenario: An account blocks sending transfer transactions
+  Scenario: An account blocks sending transactions of a given transaction type
```
Comment (Wayonb) [inline integrationtests/src/test/resources/io/nem/symbol/account_restriction/prevent_sending_transactions_by_type.feature:19]: Multiple scenarios

File: integrationtests/src/test/resources/io/nem/symbol/account_restriction/prevent_sending_transactions_by_type.feature [hunk 5]
```diff
@@ -8,85 +8,56 @@ Feature: Prevent sending transactions by type
...
-    Then Bobby should receive 1 of asset "cat.currency"
-    And Alex "cat.currency" balance should decrease by 1 units
-
   Scenario: An account only allows sending transactions of a given transaction type
     Given Alex only allows sending transactions of type:
       | TRANSFER           |
-      | NAMESPACE_REGISTRATION |
-    When Alex registers new namespace alexexp
-    Then Alex should become the owner of the new namespace alexexp
+      | REGISTER_NAMESPACE |
```
Comment (Wayonb) [inline integrationtests/src/test/resources/io/nem/symbol/account_restriction/prevent_sending_transactions_by_type.feature:35]: same here

File: integrationtests/src/test/resources/io/nem/symbol/account_restriction/prevent_sending_transactions_by_type.feature (hunk 3 above)
Comment (ravitej-aluru) [reply integrationtests/src/test/resources/io/nem/symbol/account_restriction/prevent_sending_transactions_by_type.feature]: Updated.

File: integrationtests/src/test/resources/io/nem/symbol/account_restriction/prevent_receiving_transactions_from_undesired_addresses.feature (hunk 1 above)
Comment (ravitej-aluru) [reply integrationtests/src/test/resources/io/nem/symbol/account_restriction/prevent_receiving_transactions_from_undesired_addresses.feature]: resolved

File: integrationtests/src/test/resources/io/nem/symbol/account_restriction/prevent_receiving_transactions_from_undesired_addresses.feature (hunk 2 above)
Comment (ravitej-aluru) [reply integrationtests/src/test/resources/io/nem/symbol/account_restriction/prevent_receiving_transactions_from_undesired_addresses.feature]: split scenarios

File: integrationtests/src/test/resources/io/nem/symbol/account_restriction/prevent_sending_transactions_by_type.feature (hunk 4 above)
Comment (ravitej-aluru) [reply integrationtests/src/test/resources/io/nem/symbol/account_restriction/prevent_sending_transactions_by_type.feature]: split scenarios

**Comment by ravitej-aluru** [comment]
Addressed comments

File: integrationtests/src/test/resources/io/nem/symbol/account_restriction/prevent_sending_transactions_by_type.feature [hunk 6]
```diff
@@ -25,75 +25,66 @@ Feature: Prevent sending transactions by type
...
-      | TRANSFER           |
-      | NAMESPACE_REGISTRATION |
-    When Alex removes TRANSFER from allowed transaction types
-    And Alex sends 1 asset "cat.currency" to Bobby
-    And Alex tries to register a namespace named "alexexp" for 10 blocks
-    Then Alex should receive a confirmation message
-    And Bobby should receive 1 of asset "cat.currency"
-    And Alex "cat.currency" balance should decrease by 1 units
-    And Alex should receive the error "FAILURE_RESTRICTIONACCOUNT_OPERATION_TYPE_PROHIBITED"
+      | TRANSFER                |
```
Comment (Wayonb) [inline integrationtests/src/test/resources/io/nem/symbol/account_restriction/prevent_sending_transactions_by_type.feature:83]: This scenario and above have the same same.  should be different names.

File: integrationtests/src/test/resources/io/nem/symbol/namespace/register_a_namespace.feature [hunk 7]
```diff
@@ -26,7 +26,7 @@ Feature: Register a namespace
   Scenario Outline: An account registers a namespace
     When Alice registers a namespace named "<name>" for <duration> block
     Then she should receive a confirmation message
-    And Alice should become the owner of the new namespace <name> for least <duration> block
+    And Alice should become the owner of the new namespace "<name>" for least <duration> block
```
Comment (Wayonb) [inline integrationtests/src/test/resources/io/nem/symbol/namespace/register_a_namespace.feature:29]: This step was updated but not the code.  Won't this cause failure?

File: integrationtests/src/test/resources/io/nem/symbol/account_restriction/prevent_sending_transactions_by_type.feature (hunk 6 above)
Comment (ravitej-aluru) [reply integrationtests/src/test/resources/io/nem/symbol/account_restriction/prevent_sending_transactions_by_type.feature:83]: :) Noticed it, but it was the same on master. So, I thought we need to sync-up nem2-scenarios anyway. However, fixed now

File: integrationtests/src/test/resources/io/nem/symbol/namespace/register_a_namespace.feature (hunk 7 above)
Comment (ravitej-aluru) [reply integrationtests/src/test/resources/io/nem/symbol/namespace/register_a_namespace.feature]: Fixed it

**Pull request merged**